designPatterns-4/src/creationalpattern/FactoryPattern.java

```
@@ -0,0 +1,113 @@
+package creationalpattern;
+
+import java.util.Scanner;
+
+public class FactoryPattern {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("choose you shop\nchicken\nmutton");
+        String choice = sc.nextLine();
```

Can you add input validation to this program? You can simply throw an exception or respond to the user that his input is invalid if he enters anything other than what's expected

---

Added try catch and custom exception